### PR TITLE
14150 apply primarytable alias (include case manual set alias) in validators

### PR DESCRIPTION
**Pull request by bscheshirwork**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | yes
| Breaks BC?    | no
| Fixed issues  | #14161, #14150

### CODE REVIEW

**Comment by cebe** [comment]
what is the difference to #14161? you do not need to open a new PR to make changes to an existing one, you can just push new commits to the same branch.

**Comment by samdark** [comment]
Seems to fail: https://travis-ci.org/yiisoft/yii2/jobs/232464830

**Comment by bscheshirwork** [comment]
> what is the difference to #14161?

@samdark is vote for private method in validators.

But i think we need discuss about method to automated assign alias to condition (i.e. in multiply joinWith)

File: framework/validators/ExistValidator.php [hunk 1]
```diff
@@ -222,7 +222,7 @@ private function applyTableAlias($query, $conditions, $alias = null)
         $prefixedConditions = [];
         foreach ($conditions as $columnName => $columnValue) {
             $prefixedColumn = "{$alias}.[[" . preg_replace(
-                    '/^' . $alias . '\.(.*)$/',
+                    '/^' . preg_quote($alias) . '\.(.*)$/',
                     "$1",
```
Comment (bscheshirwork) [inline framework/validators/ExistValidator.php]: `\ + * ? [ ^ ] $ ( ) { } = ! < > | : -`
Ok, the alias may consist `-`

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**